phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryWithOffsetIT.java

```diff
@@ -43,191 +44,448 @@
+            String failureSql = "SELECT count(*) FROM " + tableName + "  OFFSET ('a', 1)";
+            try {
+                conn.createStatement().execute(failureSql);
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+                return;
+            }
+            fail("Should not have allowed partial Key RVC Offset");
+        }
+
+
+        //Test RVC Offset must have an entry for every one in the pk
+        @Test
+        public void testRVCOffsetMoreThanKey() throws SQLException {
+            String failureSql = "SELECT count(*) FROM " + tableName + "  OFFSET ('a', 1, 2, 3)";
+            try {
+                conn.createStatement().execute(failureSql);
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+                return;
+            }
+            fail("Should not have allowed more than pk columns in Key RVC Offset");
+        }
+
+        //I think normal filtering where clauses should be reasonable but
+        //should this be allowed in point lookups? This is actually hard as we have a  bunch of special case code for this
+        //First step is probably error, can open a new JIRA for support point gets with this
+        //This also means if we dont' support normal where clauses we won't support multi-tenant tables probably not good enough
+
+        //Positive test i think salting should work
+        //Test reverse scan is not allowed i guess?
+
+        //Test RVC Offset must be a literal, cannot have column reference
+        @Test
+        public void testSimpleRVCOffsetLookup() throws SQLException {
+
+            String dataTableName = "T_" + generateUniqueName();
+            String dataTableDDL = "CREATE TABLE " + dataTableName + " (k1 TINYINT NOT NULL,\n" + "k2 TINYINT NOT NULL,\n"
+                    + "k3 TINYINT NOT NULL,\n" + "v1 INTEGER,\n"
+                    + "CONSTRAINT pk PRIMARY KEY (k1, k2, k3)) ";
+
+            conn.createStatement().execute(dataTableDDL);
+            conn.commit();
+
+
+            String upsertDML = String.format("UPSERT INTO %s VALUES(?,?,?,?)",dataTableName);
+
+            int nRows = 0;
+            PreparedStatement ps = conn.prepareStatement(upsertDML);
+            for(int k1 =0; k1 < 4; k1++) {
+                ps.setInt(1,k1);
+                for(int k2 =0; k2 < 4; k2++) {
+                    ps.setInt(2,k2);
+                    for(int k3 =0; k3 < 4; k3++) {
+                        ps.setInt(3,k3);
+                        ps.setInt(4,nRows);
+                        int result = ps.executeUpdate();
+                        assertEquals(1,result);
+                        nRows++;
+                    }
+                }
+            }
+            conn.commit();
+
+            //TODO: The Where clause here is currently a hack so that it goes into WhereOptimizer.pushRowKeyToScan method need to when the offset is present always go through that path
```

Comment:
Do you man we need to specify a where clause to get the RVC offset to execute?

Reply:
Right now in the code would only go into pushRowKeyToScan if the where clause is not null.  I intend to put in logic to go through that path without a where clause if a rvc offset is present.  I’m undecided on the best approach to do so.